coreos-koji-tagger/coreos_koji_tagger.py

```diff
@@ -197,15 +225,16 @@ def __init__(self):
                                         'GITHUB_REPO_BRANCH',
                                         DEFAULT_GITHUB_REPO_BRANCH)
         self.koji_user         = COREOS_KOJI_USER
+        self.koji_client       = koji.ClientSession(KOJI_SERVER_URL)
 
-        # If a keytab was specified let's use it
+        # If a keytab was specified let's try to auth.
         self.keytab_file = os.getenv('COREOS_KOJI_TAGGER_KEYTAB_FILE')
+        if self.keytab_file and not os.path.exists(self.keytab_file):
+            raise Exception("The specified keytab file "
+                            "does not exist: %s" % self.keytab_file)
         if self.keytab_file:
-            if os.path.exists(self.keytab_file):
-                self.kinit()
-            else:
-                raise Exception("The specified keytab file "
-                                "does not exist: %s" % self.keytab_file)
+                principal = self.find_principal_from_keytab()
+                self.koji_client.gssapi_login(principal, self.keytab_file)
         else:
             logger.info('No keytab file defined in '
```

Comment:
Is there ever a case where we want to run this without a keytab file being defined & existing?

Reply:
yes. It's usefull for running in a read-only/dry-run mode for testing. The following log line says: `Will not attempt koji write operations`